Do not check for pcre.h. php-oauth does not use pcre.h; it uses the pcre
functionality built into php, which uses pcre.h prior to php 7.3 and
pcre2.h in php 7.3 and later.
https://github.com/php/pecl-web_services-oauth/commit/2e02631fa424639b10d105ff47e9cf650403cef7
--- config.m4.orig	2020-09-18 02:16:55.000000000 -0500
+++ config.m4	2021-11-26 10:37:00.000000000 -0600
@@ -10,7 +10,6 @@
   PHP_NEW_EXTENSION(oauth, oauth.c provider.c, $ext_shared)
   CFLAGS="$CFLAGS -Wall -g"
 
-  AC_CHECK_HEADER(pcre.h, , [AC_MSG_ERROR([Couldn't find pcre.h, try installing the libpcre development/headers package])])
 
   AS_IF([test "x$with_curl" != "xno"],
       [
